lighthouse-core/gather/gatherers/link-elements.js

```diff
@@ -147,6 +154,29 @@ class LinkElements extends Gatherer {
 
     return linkElements;
   }
+
+  /**
+   * @param {LH.Gatherer.PassContext} passContext
+   * @param {LH.Gatherer.LoadData} loadData
+   * @return {Promise<LH.Artifacts['LinkElements']>}
+   */
+  async afterPass(passContext, loadData) {
+    return await this._getArtifact(passContext, loadData.networkRecords);
+  }
+
+  /**
+   * @param {LH.Gatherer.FRTransitionalContext<'DevtoolsLog'>} passContext
+   * @return {Promise<LH.Artifacts['LinkElements']>}
+   */
+  async getArtifact(passContext) {
+    const context = {
+      computedCache: passContext.computedCache,
+      settings: passContext.settings,
+      options: {},
```

Comment:
In the first attempt at sharing, I believe a primary objection was the references to these other pieces of state that computed artifacts have. 

There's no reason computed artifacts actually need `settings` or `options` it's just an artifact (heh) of the fact that audits always have access to this information. If we remove both of those from the typedefs (which we should to make it more generic and lower the burden of additional context here), context can be used directly without this intermediate object.

We only have two computed artifacts to fix to remove this requirement.

lighthouse-core/computed/metrics/timing-summary.js:34:74
lighthouse-core/computed/resource-summary.js:56:53 (which was just a minor mistake in the first place)

Reply:
Actually it's more than two. Timing summary needs `settings` to request a bunch of lantern metrics, which use `settings.throttlingMethod`. So we need to fix `LanternMetric` as well.

Reply:
> lighthouse-core/computed/metrics/timing-summary.js:34:74

this one's a pretty serious bug that luckily we don't end up triggering. If someone requested timing-summary with the same trace but once with `mobileSlow4G` and once with `desktopDense4G`, they'd get the same answers back because caching is only on the trace and devtoolsLog :/

The intention was definitely to only pass along the `computedCache`, but the type should have been more prescriptive that that was the case.